src/server/api/routers/book.ts

```diff
@@ -1,27 +1,203 @@
 import { z } from "zod";
+import { createTRPCRouter, protectedProcedure } from "src/server/api/trpc";
 
-import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+export interface Section {
+  id?: string;
+  title: string;
+  content: string;
+}
+
+export interface Chapter {
+  id?: string;
+  title: string;
+  sections: { title: string; content: string }[];
+  bookId?: string;
+}
+
+export interface Book {
+  id: string;
+  title: string;
+  author: string;
+  chapters?: Chapter[];
+}
 
 export const bookRouter = createTRPCRouter({
   create: protectedProcedure
-    .input(z.object({ title: z.string().min(1), author: z.string().min(1) }))
+    .input(
+      z.object({
+        title: z.string().min(1, "Title is required"),
+        author: z.string().min(1, "Author is required"),
+        chapters: z.array(
+          z.object({
+            title: z.string().min(1, "Chapter title is required"),
+            sections: z.array(
+              z.object({
+                title: z.string().min(1, "Section title is required"),
+                content: z.string().min(1, "Section content is required"),
+              }),
+            ),
+          }),
+        ),
+      }),
+    )
     .mutation(async ({ ctx, input }) => {
-      // simulate a slow db call
-      await new Promise((resolve) => setTimeout(resolve, 1000));
+      // Simulate a slow db call if necessary
+      // await new Promise((resolve) => setTimeout(resolve, 1000));
 
-      return ctx.db.book.create({
+      // Create the book with chapters and sections
+      const book = await ctx.db.book.create({
         data: {
           title: input.title,
           author: input.author,
-          createdById: ctx.session.user.id,
+          createdBy: { connect: { id: ctx.session.user.id } },
+          chapters: {
+            create: input.chapters.map((chapter: Chapter) => ({
```

Comment:
yeah this is not needed, will be inferred
```suggestion
            create: input.chapters.map((chapter) => ({
```